src/include/main/db_main.h

```diff
@@ -395,18 +395,28 @@ class DBMain {
       if (use_replication_) {
         NOISEPAGE_ASSERT(use_messenger_, "Replication uses the messenger subsystem.");
         NOISEPAGE_ASSERT(use_logging_, "Replication uses logging.");
-        replication_manager = std::make_unique<replication::ReplicationManager>(
-            messenger_layer->GetMessenger(), network_identity_, replication_port_, replication_hosts_path_,
-            common::ManagedPointer(empty_buffer_queue));
+        if (network_identity_ == "primary") {
```

Comment:
We should probably extract `"primary"` to a constant somewhere, but I don't think it's a huge deal (If you haven't been able to tell, I'm slightly obsessed with constants, so feel free to disregard this)

Reply:
tada